### DISCUSSION TITLE
Using rewire for testing Non-exported functions

### DISCUSSION THREAD

**Discussion by tieje** [Hellenic/react-hexgrid#92] [Ideas]
[rewire](https://github.com/jhnns/rewire) can be used to test non-exported functions.
This idea came up from [PR #89](https://github.com/Hellenic/react-hexgrid/pull/89).
I've never used it before, but it looks like it's been updated in the past year.
It would be nice is someone could look into the viability of rewire.

**Comment by jannikbuschke** [comment]
I am not sure if we actually want to test internals. Could be useful, however if we want to refactor internals and need to rewrite tests this can also be annoying. In my experience its more important to test the public api and behavior.

**Reply by tieje** [reply]
Yeah, it's probably for the best to hold off. This is low priority.
